03-Bash/kernel_git/kernel_stats.sh

```diff
@@ -10,3 +10,25 @@ if [[( "$#" -ne 1 ) || ( "$1" = "--help" )]]; then
   stats_help
   exit 1
 fi
+
+# Quantity of files with extention .c.
+echo Quantity of .c files in diretory
+ls -lr | grep --count \.c
+# Quantity of files with extention .cpp.
+echo Quantity of .cpp files in diretory
+ls -lr | grep --count \.cpp
+# Quantity of files with extention .py.
+echo Quantity of .py files in diretory
+ls -lr | grep --count \.py
+
+# Quantity of commits with word 'revert'.
+echo Quantity of commits with word "\'revert\'"
+git rev-list HEAD --count --grep='revert'
+
+# Lines for each author are in .py files.
+py_files=$(find $1 -type f | grep "\.py")
+for pfile in "${py_files[@]}";
+do
+echo "file: $pfile"
+git blame --line-porcelain ${pfile} | sed -n 's/^author //p' | sort | uniq -c | sort -rn
```

Comment:
Statistic about all files together is required.

Reply:
fixed with find utility